plugins/woocommerce/src/Blocks/Utils/ProductAvailabilityUtils.php

```diff
@@ -0,0 +1,48 @@
+<?php
+declare(strict_types=1);
+namespace Automattic\WooCommerce\Blocks\Utils;
+
+use Automattic\WooCommerce\Blocks\Templates\ProductStockIndicator;
+use Automattic\WooCommerce\Enums\ProductType;
+/**
+ * Utility functions for product availability.
+ */
+class ProductAvailabilityUtils {
+
+	/**
+	 * Get product availability information.
+	 *
+	 * @param \WC_Product $product Product object.
+	 * @return string[] The product availability class and text.
+	 */
+	public static function get_product_availability( $product ) {
+		$product_availability = array(
+			'availability' => '',
+			'class'        => '',
+		);
+
+		if ( ! $product ) {
+			return $product_availability;
+		}
+
+		// If the product is a variable product, check if it has any available variations.
+		// We will show a custom availability message if it does.
+		if ( $product->get_type() === ProductType::VARIABLE ) {
+			$available_variations = $product->get_available_variations();
+			if ( empty( $available_variations ) && false !== $available_variations ) {
+				$product_availability['availability'] = __( 'This product is currently out of stock and unavailable.', 'woocommerce' );
+				$product_availability['class']        = 'out-of-stock';
```

Comment:
We set the product availability class to `out-of-stock` for consistency with the variable product page. We assume it's out of stock [here](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/templates/single-product/add-to-cart/variable.php#L32).